tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -5925,6 +5958,33 @@ SpirvEmitter::doCXXOperatorCallExpr(const CXXOperatorCallExpr *expr,
     }
   }
 
+  SourceRange range =
+      (rangeOverride != SourceRange()) ? rangeOverride : expr->getSourceRange();
+
+  { // Handle ResourceDescriptorHeap and SamplerDescriptorHeap.
+    if (isDescriptorHeap(expr)) {
+      const Expr *baseExpr = nullptr;
+      const Expr *indexExpr = nullptr;
+      getDescriptorHeapOperands(expr, &baseExpr, &indexExpr);
+
+      const Expr *ParentExpr = cast<CastExpr>(parentMap->getParent(expr));
```

Comment:
Is it possible for the parent to be nullptr? It looks like it's legal for descriptor heaps to be used in global vars, can you add a test for that?

Reply:
Good point. Added a test and fix for static global variables.
For external globals, it silently ignore the warning (while DXIL fails).
Seems like there is an issue for this issue: https://github.com/microsoft/DirectXShaderCompiler/issues/3950
Thanks!